lib/grape/request.rb

```diff
@@ -15,6 +15,8 @@ def initialize(env, **options)
 
     def params
       @params ||= build_params
+    rescue EOFError
+      raise Grape::Exceptions::InvalidMessageBody, 'multipart/form-data'
```

Comment:
I am concerned that this is hardcoding `multipart/form-data`, input doesn't have to be this way, right?

Reply:
I don't know when Rack would raise this error other than when given a `multipart/form-data`, but I'm not a Rack expert.  I've pushed a commit to use the object's `content_type` method instead.

This error class seemed the most appropriated, but it requires a content type.  I'm happy to create a different error class if that would be preferable.

Reply:
I'm ok with reusing this error, it makes sense to me.

Reply:
On a second thought, should we create a EmptyMessageBody error?